compiler/src/iree/compiler/Codegen/Common/PolynomialApproximationPass.cpp

```diff
@@ -29,28 +29,64 @@ namespace {
 class PolynomialApproximationPass final
     : public impl::PolynomialApproximationPassBase<
           PolynomialApproximationPass> {
+
+public:
+  using impl::PolynomialApproximationPassBase<
+      PolynomialApproximationPass>::PolynomialApproximationPassBase;
+
   void runOnOperation() override {
+    using PatternFunction = std::function<decltype(populateExpandTanPattern)>;
+    std::map<std::string, PatternFunction> patternMap = {
+        {"tan", populateExpandTanPattern},
+        {"sinh", populateExpandSinhPattern},
+        {"cosh", populateExpandCoshPattern},
+        {"asinh", populateExpandAsinhPattern},
+        {"acosh", populateExpandAcoshPattern},
+        {"atanh", populateExpandAtanhPattern},
+        {"powf", populateExpandPowFPattern},
+        {"fpowi", populateExpandFPowIPattern},
+        {"erf",
+         [&](RewritePatternSet &mathPatterns) {
+           this->populateErfPattern(mathPatterns);
+         }},
+    };
+
     RewritePatternSet mathPatterns(&getContext());
-    populateExpandTanPattern(mathPatterns);
-    populateExpandSinhPattern(mathPatterns);
-    populateExpandCoshPattern(mathPatterns);
-    populateExpandAsinhPattern(mathPatterns);
-    populateExpandAcoshPattern(mathPatterns);
-    populateExpandAtanhPattern(mathPatterns);
-    populateExpandPowFPattern(mathPatterns);
-    populateExpandFPowIPattern(mathPatterns);
 
+    std::vector<StringRef> noOps = splitByComma(noApproxOps);
```

Comment:
Don't use `std::vector` in llvm code: https://llvm.org/docs/CodingStandards.html#c-standard-library and https://llvm.org/docs/ProgrammersManual.html#llvm-adt-smallvector-h